File: fegen-example-gradle/kotlin-api/src/main/kotlin/de/materna/fegen/example/gradle/kotlin/api/controller/TestRestControllerClient.kt
package de.materna.fegen.example.gradle.kotlin.api.controller

import com.fasterxml.jackson.core.type.TypeReference
import de.materna.fegen.example.gradle.kotlin.api.ComplexTestPojo
import de.materna.fegen.example.gradle.kotlin.api.CyclicTestPojoA
import de.materna.fegen.example.gradle.kotlin.api.PrimitiveTestEntity
import de.materna.fegen.example.gradle.kotlin.api.PrimitiveTestEntityBase
import de.materna.fegen.example.gradle.kotlin.api.PrimitiveTestEntityDto
import de.materna.fegen.example.gradle.kotlin.api.PrimitiveTestPojo
import de.materna.fegen.example.gradle.kotlin.api.RecursiveTestPojo
import de.materna.fegen.runtime.ApiHateoasList
import de.materna.fegen.runtime.ApiHateoasPage
import de.materna.fegen.runtime.PagedItems
import de.materna.fegen.runtime.RequestAdapter
import de.materna.fegen.runtime.appendParams
import de.materna.fegen.runtime.isEndpointCallAllowed
import java.time.LocalDate
import java.time.LocalDateTime
import kotlin.Boolean
import kotlin.Int
import kotlin.Long
import kotlin.String
import kotlin.Suppress
import kotlin.collections.List

class TestRestControllerClient(
    private val requestAdapter: RequestAdapter
) {
    @Suppress("UNUSED")
    suspend fun cyclicPojo(body: CyclicTestPojoA): CyclicTestPojoA {
        val url = """/api/custom/testController/cyclicPojo""".appendParams()
        return requestAdapter.doSingleRequestWithoutReturnValueTransformation<CyclicTestPojoA,
                CyclicTestPojoA>(
            url = url,
            method = "POST",
            body = body,
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isCyclicPojoAllowed(): Boolean {
        val url = """/api/custom/testController/cyclicPojo"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun mixed(
        body: PrimitiveTestEntityBase,
        int32: Int,
        long64: Long
    ): PrimitiveTestEntity {
        val url = """/api/custom/testController/mixedCreate/$int32""".appendParams(
            "long64" to long64
        )
        return requestAdapter.doSingleRequest<PrimitiveTestEntity, PrimitiveTestEntityDto,
                PrimitiveTestEntityBase>(
            url = url,
            method = "POST",
            body = body,
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isMixedAllowed(int32: Int): Boolean {
        val url = """/api/custom/testController/mixedCreate/$int32"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun noBody(int32: Int, long64: Long): PrimitiveTestEntity {
        val url = """/api/custom/testController/noBodyCreate/$int32""".appendParams(
            "long64" to long64
        )
        return requestAdapter.doSingleRequest<PrimitiveTestEntity, PrimitiveTestEntityDto>(
            url = url,
            method = "POST",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isNoBodyAllowed(int32: Int): Boolean {
        val url = """/api/custom/testController/noBodyCreate/$int32"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun noPathVariable(body: PrimitiveTestEntityBase, long64: Long): PrimitiveTestEntity {
        val url = """/api/custom/testController/noPathVariableCreate""".appendParams(
            "long64" to long64
        )
        return requestAdapter.doSingleRequest<PrimitiveTestEntity, PrimitiveTestEntityDto,
                PrimitiveTestEntityBase>(
            url = url,
            method = "POST",
            body = body,
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isNoPathVariableAllowed(): Boolean {
        val url = """/api/custom/testController/noPathVariableCreate"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun noRequestParam(body: PrimitiveTestEntityBase, int32: Int): PrimitiveTestEntity {
        val url = """/api/custom/testController/noRequestParamCreate/$int32""".appendParams()
        return requestAdapter.doSingleRequest<PrimitiveTestEntity, PrimitiveTestEntityDto,
                PrimitiveTestEntityBase>(
            url = url,
            method = "POST",
            body = body,
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isNoRequestParamAllowed(int32: Int): Boolean {
        val url = """/api/custom/testController/noRequestParamCreate/$int32"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun pathVariable(
        int32: Int,
        long64Custom: Long,
        intMinusBillion: Int,
        stringText: String,
        booleanTrue: Boolean,
        dateCustom: LocalDate
    ): PrimitiveTestEntity {
        val url =
                """/api/custom/testController/pathVariableCreate/$int32/$long64Custom/$intMinusBillion/$stringText/$booleanTrue/$dateCustom""".appendParams()
        return requestAdapter.doSingleRequest<PrimitiveTestEntity, PrimitiveTestEntityDto>(
            url = url,
            method = "POST",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isPathVariableAllowed(
        int32: Int,
        long64Custom: Long,
        intMinusBillion: Int,
        stringText: String,
        booleanTrue: Boolean,
        dateCustom: LocalDate
    ): Boolean {
        val url =
                """/api/custom/testController/pathVariableCreate/$int32/$long64Custom/$intMinusBillion/$stringText/$booleanTrue/$dateCustom"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun pojoAsBodyAndListReturnValue(body: ComplexTestPojo): List<ComplexTestPojo> {
        val url = """/api/custom/testController/pojoAsBodyAndListReturnValue""".appendParams()
        return requestAdapter.doListRequestSimple(
            url = url,
            method = "POST",
            body = body,
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isPojoAsBodyAndListReturnValueAllowed(): Boolean {
        val url = """/api/custom/testController/pojoAsBodyAndListReturnValue"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun pojoAsBodyAndReturnValue(body: ComplexTestPojo): ComplexTestPojo {
        val url = """/api/custom/testController/pojoAsBodyAndSingleReturnValue""".appendParams()
        return requestAdapter.doSingleRequestWithoutReturnValueTransformation<ComplexTestPojo,
                ComplexTestPojo>(
            url = url,
            method = "POST",
            body = body,
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isPojoAsBodyAndReturnValueAllowed(): Boolean {
        val url = """/api/custom/testController/pojoAsBodyAndSingleReturnValue"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun pojoListAsBody(body: List<PrimitiveTestPojo>): List<PrimitiveTestPojo> {
        val url = """/api/custom/testController/pojoListAsBody""".appendParams()
        return requestAdapter.doListRequestSimple(
            url = url,
            method = "POST",
            body = body,
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isPojoListAsBodyAllowed(): Boolean {
        val url = """/api/custom/testController/pojoListAsBody"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun pojosAsReturnValue(): List<PrimitiveTestPojo> {
        val url = """/api/custom/testController/pojosAsReturnValue""".appendParams()
        return requestAdapter.doListRequestSimple(
            url = url,
            method = "GET",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isPojosAsReturnValueAllowed(): Boolean {
        val url = """/api/custom/testController/pojosAsReturnValue"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "GET", url)
    }

    @Suppress("UNUSED")
    suspend fun recursivePojo(body: RecursiveTestPojo): RecursiveTestPojo {
        val url = """/api/custom/testController/recursivePojo""".appendParams()
        return requestAdapter.doSingleRequestWithoutReturnValueTransformation<RecursiveTestPojo,
                RecursiveTestPojo>(
            url = url,
            method = "POST",
            body = body,
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isRecursivePojoAllowed(): Boolean {
        val url = """/api/custom/testController/recursivePojo"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun requestParam(
        int32: Int,
        long64Custom: Long,
        optionalIntNull: Int?,
        optionalIntBillion: Int?,
        intMinusBillion: Int,
        stringText: String,
        booleanTrue: Boolean,
        dateCustom: LocalDate,
        dateTime2000_1_1_12_30: LocalDateTime?
    ): PrimitiveTestEntity {
        val url = """/api/custom/testController/requestParamCreate""".appendParams(
            "int32" to int32,
            "long64Custom" to long64Custom,
            "optionalIntNull" to optionalIntNull,
            "optionalIntBillion" to optionalIntBillion,
            "intMinusBillion" to intMinusBillion,
            "stringText" to stringText,
            "booleanTrue" to booleanTrue,
            "dateCustom" to dateCustom,
            "dateTime2000_1_1_12_30" to dateTime2000_1_1_12_30
        )
        return requestAdapter.doSingleRequest<PrimitiveTestEntity, PrimitiveTestEntityDto>(
            url = url,
            method = "POST",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isRequestParamAllowed(): Boolean {
        val url = """/api/custom/testController/requestParamCreate"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun responseBody(body: PrimitiveTestEntityBase): PrimitiveTestEntity {
        val url = """/api/custom/testController/requestBodyCreate""".appendParams()
        return requestAdapter.doSingleRequest<PrimitiveTestEntity, PrimitiveTestEntityDto,
                PrimitiveTestEntityBase>(
            url = url,
            method = "POST",
            body = body,
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isResponseBodyAllowed(): Boolean {
        val url = """/api/custom/testController/requestBodyCreate"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "POST", url)
    }

    @Suppress("UNUSED")
    suspend fun returnBoolean(): Boolean {
        val url = """/api/custom/testController/returnBoolean""".appendParams()
        return requestAdapter.doSingleRequestWithoutReturnValueTransformation<Boolean>(
            url = url,
            method = "GET",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isReturnBooleanAllowed(): Boolean {
        val url = """/api/custom/testController/returnBoolean"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "GET", url)
    }

    @Suppress("UNUSED")
    suspend fun returnInteger(): Int {
        val url = """/api/custom/testController/returnInteger""".appendParams()
        return requestAdapter.doSingleRequestWithoutReturnValueTransformation<Int>(
            url = url,
            method = "GET",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isReturnIntegerAllowed(): Boolean {
        val url = """/api/custom/testController/returnInteger"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "GET", url)
    }

    @Suppress("UNUSED")
    suspend fun returnList(): List<PrimitiveTestEntity> {
        val url = """/api/custom/testController/returnList""".appendParams()
        return requestAdapter.doListRequest(
            url = url,
            method = "GET",
            embeddedPropName = "primitiveTestEntities",
            ignoreBasePath = true,
            type = object : TypeReference<ApiHateoasList<PrimitiveTestEntityDto,
                PrimitiveTestEntity>>() {}
        )
    }

    @Suppress("UNUSED")
    suspend fun isReturnListAllowed(): Boolean {
        val url = """/api/custom/testController/returnList"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "GET", url)
    }

    @Suppress("UNUSED")
    suspend fun returnPaged(
        page: Int? = null,
        size: Int? = null,
        sort: String? = null
    ): PagedItems<PrimitiveTestEntity> {
        val url = """/api/custom/testController/returnPaged""".appendParams()
        return requestAdapter.doPageRequest(
            url = url,
            method = "GET",
            embeddedPropName = "primitiveTestEntities",
            page = page,
            size = size,
            sort = sort,
            ignoreBasePath = true,
            type = object : TypeReference<ApiHateoasPage<PrimitiveTestEntityDto,
                PrimitiveTestEntity>>() {}
        )
    }

    @Suppress("UNUSED")
    suspend fun isReturnPagedAllowed(): Boolean {
        val url = """/api/custom/testController/returnPaged"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "GET", url)
    }

    @Suppress("UNUSED")
    suspend fun returnString(): String {
        val url = """/api/custom/testController/returnString""".appendParams()
        return requestAdapter.doSingleRequestWithoutReturnValueTransformation<String>(
            url = url,
            method = "GET",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isReturnStringAllowed(): Boolean {
        val url = """/api/custom/testController/returnString"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "GET", url)
    }

    @Suppress("UNUSED")
    suspend fun returnStringList(): List<String> {
        val url = """/api/custom/testController/returnStringList""".appendParams()
        return requestAdapter.doListRequestSimple(
            url = url,
            method = "GET",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isReturnStringListAllowed(): Boolean {
        val url = """/api/custom/testController/returnStringList"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "GET", url)
    }

    @Suppress("UNUSED")
    suspend fun returnVoid() {
        val url = """/api/custom/testController/returnVoid""".appendParams()
        return requestAdapter.doVoidRequest(
            url = url,
            method = "GET",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isReturnVoidAllowed(): Boolean {
        val url = """/api/custom/testController/returnVoid"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "GET", url)
    }

    @Suppress("UNUSED")
    suspend fun securedEndpoint(pathParam: String, queryParam: Long): String {
        val url = """/api/custom/testController/securedEndpoint/$pathParam""".appendParams(
            "queryParam" to queryParam
        )
        return requestAdapter.doSingleRequestWithoutReturnValueTransformation<String>(
            url = url,
            method = "PUT",
            ignoreBasePath = true
        )
    }

    @Suppress("UNUSED")
    suspend fun isSecuredEndpointAllowed(pathParam: String): Boolean {
        val url = """/api/custom/testController/securedEndpoint/$pathParam"""
        return isEndpointCallAllowed(requestAdapter.fetchAdapter, "//api", "PUT", url)
    }
}

File: fegen-example-gradle/web-api/src/controller/TestRestControllerClient.ts
import {CyclicTestPojoA, PrimitiveTestEntity, PrimitiveTestEntityNew, ComplexTestPojo, PrimitiveTestPojo, RecursiveTestPojo} from "../Entities";
import {RequestAdapter, stringHelper, isEndpointCallAllowed, Items, ApiHateoasObjectBase, apiHelper, PagedItems, ApiHateoasObjectReadMultiple} from "@materna-se/fegen-runtime";

export class TestRestControllerClient {

    private readonly requestAdapter: RequestAdapter;
    
    constructor(requestAdapter: RequestAdapter) {
        this.requestAdapter = requestAdapter;
    }
    
    public async cyclicPojo(body: CyclicTestPojoA): Promise<CyclicTestPojoA>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/cyclicPojo`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(body),
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as CyclicTestPojoA;
        
        
    }
    
    public isCyclicPojoAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/cyclicPojo`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async mixed(int32: number, body: PrimitiveTestEntityNew, long64: number): Promise<PrimitiveTestEntity>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/mixedCreate/${int32}`;
    
        const params = {long64};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(body),
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as PrimitiveTestEntity;
        return responseObj;
        
        
    }
    
    public isMixedAllowed(int32: number): Promise<boolean> {
        const url = `/api/custom/testController/mixedCreate/${int32}`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async noBody(int32: number, long64: number): Promise<PrimitiveTestEntity>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/noBodyCreate/${int32}`;
    
        const params = {long64};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as PrimitiveTestEntity;
        return responseObj;
        
        
    }
    
    public isNoBodyAllowed(int32: number): Promise<boolean> {
        const url = `/api/custom/testController/noBodyCreate/${int32}`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async noPathVariable(body: PrimitiveTestEntityNew, long64: number): Promise<PrimitiveTestEntity>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/noPathVariableCreate`;
    
        const params = {long64};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(body),
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as PrimitiveTestEntity;
        return responseObj;
        
        
    }
    
    public isNoPathVariableAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/noPathVariableCreate`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async noRequestParam(int32: number, body: PrimitiveTestEntityNew): Promise<PrimitiveTestEntity>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/noRequestParamCreate/${int32}`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(body),
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as PrimitiveTestEntity;
        return responseObj;
        
        
    }
    
    public isNoRequestParamAllowed(int32: number): Promise<boolean> {
        const url = `/api/custom/testController/noRequestParamCreate/${int32}`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async pathVariable(int32: number, long64Custom: number, intMinusBillion: number, stringText: string, booleanTrue: boolean, dateCustom: string): Promise<PrimitiveTestEntity>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/pathVariableCreate/${int32}/${long64Custom}/${intMinusBillion}/${stringText}/${booleanTrue}/${dateCustom}`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as PrimitiveTestEntity;
        return responseObj;
        
        
    }
    
    public isPathVariableAllowed(int32: number, long64Custom: number, intMinusBillion: number, stringText: string, booleanTrue: boolean, dateCustom: string): Promise<boolean> {
        const url = `/api/custom/testController/pathVariableCreate/${int32}/${long64Custom}/${intMinusBillion}/${stringText}/${booleanTrue}/${dateCustom}`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async pojoAsBodyAndListReturnValue(body: ComplexTestPojo): Promise<ComplexTestPojo[]>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/pojoAsBodyAndListReturnValue`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(body),
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as ComplexTestPojo[];
        
        
    }
    
    public isPojoAsBodyAndListReturnValueAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/pojoAsBodyAndListReturnValue`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async pojoAsBodyAndReturnValue(body: ComplexTestPojo): Promise<ComplexTestPojo>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/pojoAsBodyAndSingleReturnValue`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(body),
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as ComplexTestPojo;
        
        
    }
    
    public isPojoAsBodyAndReturnValueAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/pojoAsBodyAndSingleReturnValue`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async pojoListAsBody(body: PrimitiveTestPojo[]): Promise<PrimitiveTestPojo[]>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/pojoListAsBody`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(body),
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as PrimitiveTestPojo[];
        
        
    }
    
    public isPojoListAsBodyAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/pojoListAsBody`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async pojosAsReturnValue(): Promise<PrimitiveTestPojo[]>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/pojosAsReturnValue`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "GET"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as PrimitiveTestPojo[];
        
        
    }
    
    public isPojosAsReturnValueAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/pojosAsReturnValue`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "GET", url);
    }
    
    public async recursivePojo(body: RecursiveTestPojo): Promise<RecursiveTestPojo>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/recursivePojo`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(body),
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as RecursiveTestPojo;
        
        
    }
    
    public isRecursivePojoAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/recursivePojo`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async requestParam(int32: number, long64Custom: number, intMinusBillion: number, stringText: string, booleanTrue: boolean, dateCustom: string, optionalIntNull?: number, optionalIntBillion?: number, dateTime2000_1_1_12_30?: string): Promise<PrimitiveTestEntity>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/requestParamCreate`;
    
        const params = {int32, long64Custom, optionalIntNull, optionalIntBillion, intMinusBillion, stringText, booleanTrue, dateCustom, dateTime2000_1_1_12_30};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as PrimitiveTestEntity;
        return responseObj;
        
        
    }
    
    public isRequestParamAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/requestParamCreate`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async responseBody(body: PrimitiveTestEntityNew): Promise<PrimitiveTestEntity>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/requestBodyCreate`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "POST",
                headers: {
                    "content-type": "application/json"
                },
                body:JSON.stringify(body),
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as PrimitiveTestEntity;
        return responseObj;
        
        
    }
    
    public isResponseBodyAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/requestBodyCreate`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "POST", url);
    }
    
    public async returnBoolean(): Promise<boolean>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/returnBoolean`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "GET"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as boolean;
        
        
    }
    
    public isReturnBooleanAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/returnBoolean`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "GET", url);
    }
    
    public async returnInteger(): Promise<number>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/returnInteger`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "GET"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as number;
        
        
    }
    
    public isReturnIntegerAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/returnInteger`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "GET", url);
    }
    
    public async returnList(): Promise<Items<PrimitiveTestEntity>>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/returnList`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "GET"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as ApiHateoasObjectBase<PrimitiveTestEntity[]>;
        const elements = ((responseObj._embedded && responseObj._embedded.primitiveTestEntities) || []).map(item => (apiHelper.injectIds(item)));
        
        return {
            items: elements,
            _links: responseObj._links
        };
        
        
    }
    
    public isReturnListAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/returnList`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "GET", url);
    }
    
    public async returnPaged(page?: number, size?: number, sort?: "id,ASC" | "id,DESC" | "booleanTrue,ASC" | "booleanTrue,DESC" | "date2000_6_12,ASC" | "date2000_6_12,DESC" | "int32,ASC" | "int32,DESC" | "intMinusBillion,ASC" | "intMinusBillion,DESC" | "long64,ASC" | "long64,DESC" | "stringText,ASC" | "stringText,DESC"): Promise<PagedItems<PrimitiveTestEntity>>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/returnPaged`;
    
        const params = {page, size, sort};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "GET"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        const responseObj = (await response.json()) as ApiHateoasObjectReadMultiple<PrimitiveTestEntity[]>;
        const elements = ((responseObj._embedded && responseObj._embedded.primitiveTestEntities) || []).map(item => (apiHelper.injectIds(item)));
        
        return {
            items: elements,
            _links: responseObj._links,
            page: responseObj.page
        };
        
        
    }
    
    public isReturnPagedAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/returnPaged`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "GET", url);
    }
    
    public async returnString(): Promise<string>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/returnString`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "GET"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as string;
        
        
    }
    
    public isReturnStringAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/returnString`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "GET", url);
    }
    
    public async returnStringList(): Promise<Items<string>>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/returnStringList`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "GET"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as Items<string>;
        
        
    }
    
    public isReturnStringListAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/returnStringList`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "GET", url);
    }
    
    public async returnVoid(): Promise<void>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/returnVoid`;
    
        const params = {};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "GET"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
    }
    
    public isReturnVoidAllowed(): Promise<boolean> {
        const url = `/api/custom/testController/returnVoid`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "GET", url);
    }
    
    public async securedEndpoint(pathParam: string, queryParam: number): Promise<string>  {
        const request = this.requestAdapter.fetchAdapter;
    
        const baseUrl = `/api/custom/testController/securedEndpoint/${pathParam}`;
    
        const params = {queryParam};
    
        const url = stringHelper.appendParams(baseUrl, params);
    
        const response = await request.fetch(
            url,
            {
                method: "PUT"
            },
            true);
    
        if(!response.ok) {
            throw response;
        }
        
        
        return (await response.json()) as string;
        
        
    }
    
    public isSecuredEndpointAllowed(pathParam: string): Promise<boolean> {
        const url = `/api/custom/testController/securedEndpoint/${pathParam}`;
        return isEndpointCallAllowed(this.requestAdapter.fetchAdapter, "//api", "PUT", url);
    }
}